test/ecto/query_test.exs

```diff
@@ -266,4 +266,13 @@ defmodule Ecto.QueryTest do
     assert excluded_query.preloads == base.preloads
     assert excluded_query.assocs == base.assocs
   end
+
+  test "fragment/1 raises  at runtime when interpolation is not a keyword list" do
+    assert_raise ArgumentError, "Only a keyword list may be interpolated as the first " <>
+                                "argument to `fragment/1` with the `^` operator.",
+    fn ->
```

Comment:
It'd be better to keep the `fn ->` on the previous line, as for now it's harder to follows `assert_raise` definition.
Maybe even better to assign message to variable before an assertion.